internal/alb/tg/targethealth.go

```diff
@@ -202,6 +202,9 @@ func (c *targetHealthController) reconcilePodConditions(ctx context.Context, tgA
 
 	for i, target := range targetsToReconcile {
 		pod := pods[i]
+		if pod == nil {
+			continue
+		}
```

Comment:
Putting this comment here because GH doesn't let me add a comment about a commit summary/description...

In this commit message, you misspelled "readiness" as "readness". Also, the commit message does not explain *what* is being fixed by the commit's contents, so it's impossible to tell from the git log/history what these code changes are meant to address.

Reply:
Thanks. Updated the commit message to be 'bug fix for edge cases not covered by pod readness gate PR #955'
It's addressing missing parts of https://github.com/kubernetes-sigs/aws-alb-ingress-controller/pull/955. Where the resolved pod from endpoint IP can be nil when pod is not found. This can happen since the endpoint resolve and reserve resolve is based on live pod data store in real time and can be inconsistent.

Reply:
I see that you added unit test for the other change, can you add one for this as well as for targetgroup_group.go changes?

Reply:
added :D